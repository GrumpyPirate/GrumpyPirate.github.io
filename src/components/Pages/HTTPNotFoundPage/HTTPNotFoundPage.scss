@import 'core';

.http-not-found-page {
  &__heading {
    margin-bottom: $paragraph-margin-bottom * 2;
  }
}
