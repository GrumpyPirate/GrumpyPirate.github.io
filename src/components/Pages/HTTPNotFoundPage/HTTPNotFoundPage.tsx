import React, { FunctionComponent } from 'react';
import styled from 'styled-components';

import Button from 'components/Button/Button';
import Page from 'components/Page/Page';
import PageContent from 'components/PageContent/PageContent';
import PageHeader from 'components/PageHeader/PageHeader';
import PageHeaderSubtitle from 'components/PageHeaderSubtitle/PageHeaderSubtitle';
import Heading from 'components/Typography/Heading/Heading';

const Title = styled(Heading)`
  margin-bottom: 2rem;
`;

const HTTPNotFoundPage: FunctionComponent = function () {
  return (
    <Page>
      <PageHeader title="404">
        <PageHeaderSubtitle>Not Found</PageHeaderSubtitle>
      </PageHeader>

      <PageContent>
        <section>
          <Title level={4} text="That page doesn&#039;t exist." />
          <Title
            level={5}
            text="&#xff08;&#x256f;&deg;&#x25a1;&deg;&#xff09;&#x256f;&#xfe35;( .o.)"
          />

          <div>
            <Button to="/">Home</Button>
          </div>
        </section>
      </PageContent>
    </Page>
  );
};

export default HTTPNotFoundPage;
