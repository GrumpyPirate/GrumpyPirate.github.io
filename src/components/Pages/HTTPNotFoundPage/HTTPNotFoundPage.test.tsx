import React from 'react';
import { shallow } from 'enzyme';

import HTTPNotFoundPage from './HTTPNotFoundPage';

describe('Components', () => {
  describe('HTTPNotFoundPage', () => {
    it('should render as expected, without crashing', () => {
      const wrapper = shallow(<HTTPNotFoundPage />);

      expect(wrapper).toMatchSnapshot();
    });
  });
});
