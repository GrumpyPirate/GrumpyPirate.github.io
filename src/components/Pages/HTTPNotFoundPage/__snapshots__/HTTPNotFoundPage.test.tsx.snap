// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components HTTPNotFoundPage should render as expected, without crashing 1`] = `
.c12 {
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  border: none;
  margin: 0;
  padding: 0;
  width: auto;
  overflow: visible;
  background: transparent;
  border-radius: 0;
  box-shadow: none;
  cursor: pointer;
  color: inherit;
  font: inherit;
  line-height: normal;
  -webkit-appearance: none;
  touch-action: manipulation;
  background-color: #dc8381;
  border: solid 1px #dc8381;
  border-radius: 2rem;
  box-shadow: none;
  color: #fff;
  display: inline-block;
  font-family: 'Open Sans',sans-serif;
  font-size: 1rem;
  font-weight: 700;
  line-height: 1.375;
  padding: 0.5rem 1.5rem;
  text-transform: uppercase;
  -webkit-transition-duration: 0.15s;
  transition-duration: 0.15s;
  -webkit-transition-property: -webkit-transform,color,background-color;
  -webkit-transition-property: transform,color,background-color;
  transition-property: transform,color,background-color;
  -webkit-transition-timing-function: ease;
  transition-timing-function: ease;
}

.c12::-moz-focus-inner {
  border: 0;
  padding: 0;
}

.c12,
.c12:hover,
.c12:focus,
.c12:active {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c12:disabled,
.c12[disabled],
.c12:disabled:hover,
.c12[disabled]:hover,
.c12:disabled:focus,
.c12[disabled]:focus,
.c12:disabled:active,
.c12[disabled]:active {
  background-color: #ccc;
  border-color: #ccc;
  color: #fff;
  cursor: not-allowed;
  pointer-events: none;
}

.c0 {
  background-color: hsl(0,0%,95%);
}

.c8 {
  padding-bottom: 1.5rem;
  padding-top: 1.5rem;
  text-align: center;
}

.c7 {
  position: relative;
  -webkit-flex: 1 0 auto;
  -ms-flex: 1 0 auto;
  flex: 1 0 auto;
  max-width: 100%;
  min-height: 1px;
  padding-right: 0.75rem;
  padding-left: 0.75rem;
}

.c2 {
  width: 100%;
  padding-right: 0.75rem;
  padding-left: 0.75rem;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: row wrap;
  -ms-flex-flow: row wrap;
  flex-flow: row wrap;
  margin-right: -0.75rem;
  margin-left: -0.75rem;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c1 {
  background-color: #302445;
  border-bottom: solid 0.375rem #dc8381;
  text-align: center;
  text-transform: uppercase;
}

.c3 {
  color: #dc8381;
  padding: 1rem 0;
  position: relative;
}

.c4 {
  font-family: 'Titillium Web',sans-serif;
  font-weight: 300;
  -webkit-letter-spacing: 0.2em;
  -moz-letter-spacing: 0.2em;
  -ms-letter-spacing: 0.2em;
  letter-spacing: 0.2em;
  line-height: 1.2;
  margin: 0 0 1rem;
  text-transform: uppercase;
  font-size: 2.475rem;
  font-size: 12vmin;
  font-weight: 200;
  margin: 0;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c4::first-letter {
  margin-left: 0.25em;
}

.c5 {
  font-family: 'Titillium Web',sans-serif;
  font-weight: 300;
  -webkit-letter-spacing: 0.2em;
  -moz-letter-spacing: 0.2em;
  -ms-letter-spacing: 0.2em;
  letter-spacing: 0.2em;
  line-height: 1.2;
  margin: 0 0 1rem;
  text-transform: uppercase;
  font-size: 1rem;
  color: rgba(242,241,243,0.5);
  margin: 0;
  padding-bottom: 1rem;
  text-transform: uppercase;
}

.c5 a {
  color: #dc8381;
}

.c5 a:hover,
.c5 a:focus {
  color: hsl(1.3000000000000114,56.5%,75.3%);
}

.c5 a:active {
  color: hsl(1.3000000000000114,56.5%,61.6%);
}

.c9 {
  font-family: 'Titillium Web',sans-serif;
  font-weight: 300;
  -webkit-letter-spacing: 0.2em;
  -moz-letter-spacing: 0.2em;
  -ms-letter-spacing: 0.2em;
  letter-spacing: 0.2em;
  line-height: 1.2;
  margin: 0 0 1rem;
  text-transform: uppercase;
  font-size: 1rem;
}

.c11 {
  font-family: 'Titillium Web',sans-serif;
  font-weight: 300;
  -webkit-letter-spacing: 0.2em;
  -moz-letter-spacing: 0.2em;
  -ms-letter-spacing: 0.2em;
  letter-spacing: 0.2em;
  line-height: 1.2;
  margin: 0 0 1rem;
  text-transform: uppercase;
  font-size: 1rem;
}

.c10 {
  margin-bottom: 2rem;
}

@media (min-width:48em) {
  .c8 {
    padding-bottom: 3rem;
    padding-top: 3rem;
  }
}

@media (min-width:48em) {
  .c7 {
    padding-right: 0.9375rem;
    padding-left: 0.9375rem;
  }
}

@media (min-width:30em) {
  .c7 {
    -webkit-flex-basis: 83.33333333333334%;
    -ms-flex-preferred-size: 83.33333333333334%;
    flex-basis: 83.33333333333334%;
    max-width: 83.33333333333334%;
  }
}

@media (min-width:48em) {
  .c7 {
    -webkit-flex-basis: 66.66666666666666%;
    -ms-flex-preferred-size: 66.66666666666666%;
    flex-basis: 66.66666666666666%;
    max-width: 66.66666666666666%;
  }
}

@media (min-width:48em) {
  .c2 {
    padding-right: 0.9375rem;
    padding-left: 0.9375rem;
  }
}

@media (min-width:48em) {
  .c6 {
    margin-right: -0.9375rem;
    margin-left: -0.9375rem;
  }
}

@media (min-width:75em) {
  .c1 {
    border-bottom-width: 0.5rem;
  }
}

@media (min-width:48em) {
  .c3 {
    padding-top: 2rem;
  }
}

@media (min-width:48em) {
  .c4 {
    font-size: 4.5rem;
  }
}

@media (min-width:48em) {
  .c5 {
    font-size: 1.5rem;
  }
}

@media (min-width:48em) {
  .c5 {
    padding-bottom: 2rem;
  }
}

@media (min-width:48em) {
  .c9 {
    font-size: 1.5rem;
  }
}

@media (min-width:48em) {
  .c11 {
    font-size: 1rem;
  }
}

<div
  class="c0"
>
  <header
    class="c1"
  >
    <div
      class="c2"
    >
      <div
        class=""
      >
        <div
          class="c3"
        >
          <h1
            class="c4"
          >
            404
          </h1>
        </div>
        <h2
          class="c5"
        >
          <div
            class="c2"
          >
            <div
              class="c6"
            >
              <div
                class="c7"
              >
                Not Found
              </div>
            </div>
          </div>
        </h2>
      </div>
    </div>
  </header>
  <div
    class="c8"
  >
    <section>
      <h4
        class="c9 c10"
      >
        That page doesn't exist.
      </h4>
      <h5
        class="c11 c10"
      >
        （╯°□°）╯︵( .o.)
      </h5>
      <div>
        <a
          class="c12"
          href="/"
        >
          Home
        </a>
      </div>
    </section>
  </div>
</div>
`;
